src/platform/context/egl.c

```diff
@@ -170,25 +272,71 @@ terminate(struct ctx *context)
  * like mesa will be able to adverise these (even though it can do EGL 1.5).
  */
 static EGLDisplay
-get_display(struct ctx *context, EGLint type, void *native)
+get_display(struct ctx *context, EGLint type, void *native, int drm_fd)
 {
-   PFNEGLGETPLATFORMDISPLAYEXTPROC getPlatformDisplayEXT;
+   /* In practice any EGL 1.5 implementation would support the EXT extension */
+   if (context->api.eglGetPlatformDisplayEXT) {
+      if (has_extension(context, "EGL_EXT_platform_device") && has_device_extension(context, "EGL_EXT_device_drm")) {
+         /*
+          * Provide the DRM fd when creating the EGLDisplay, so that the
+          * EGL implementation can make any necessary DRM calls using the
+          * same fd as the application.
+          */
+         EGLint attribs[] = {
+            EGL_DRM_MASTER_FD_EXT, drm_fd,
+            EGL_NONE
+         };
 
-   /* Initialize extensions to those of the NULL display, for has_extension */
-   context->extensions = EGL_CALL(eglQueryString(NULL, EGL_EXTENSIONS));
+         return context->api.eglGetPlatformDisplayEXT(type, native, attribs);
+      }
 
-   /* In practise any EGL 1.5 implementation would support the EXT extension */
-   if (has_extension(context, "EGL_EXT_platform_base")) {
-      PFNEGLGETPLATFORMDISPLAYEXTPROC getPlatformDisplayEXT =
-            (void *) eglGetProcAddress("eglGetPlatformDisplayEXT");
-      if (getPlatformDisplayEXT)
-          return getPlatformDisplayEXT(type, native, NULL);
+      return context->api.eglGetPlatformDisplayEXT(type, native, NULL);
    }
```

Comment:
No I don't think so, all the checks are still there just moved to a different place.